en/complete_vehicles/atl_mantis_edu.md

```diff
@@ -0,0 +1,211 @@
+- Autopilot/FMU based on STM32F7.
+- Gimbal controller.
+- Camera system on a chip with WiFi (forwards MAVLink messages from autopilot and gimbal via Wifi)
+- Downward-facing flow module with laser range sensor.
+
+## Update Mantis EDU
+
+The files to update the Mantis can currently be found on [GoogleDrive](https://drive.google.com/drive/u/1/folders/1LXW66IBGc_SgGLmk0ltl3R8IQYYVUvtz).
+
+Files include:
+- **firmware.bin**: Camera firmware
+- **gimbal.yuneec**: Gimbal firmware
+- **autopilot.px4**: Autopilot firmware
+- **update.lzo**: RC firmware
+
+### Update Camera
+
+1. Place camera image **firmware.bin** on the SD card.
+2. Power up the drone.
+   It will reboot and update automatically.
+3. Wait until the WiFi re-appears and re-connect.
+
+Note: if the file has been removed from the SD card, the camera has successfully been updated.
+
+### Update Gimbal
+
+1. Place the gimbal image **gimbal.yuneec** on the SD card.
+2. Power up the drone.
+   The gimbal will be automatically updated.
+3. After the update, you might hear a faint beep as the gimbal starts up again.
+
+:::note
+The file will be renamed to **gimbal.yuneec.updated** if the update was successful.
+If not, you should find a file **gimbal_update.log** with more details.
+:::
+
+### Update Autopilot (PX4)
+
+1. Place the **autopilot.px4** file on the SD card.
+2. Power up the drone.
+   The autopilot will be automatically updated.
+3. After the update, you should hear PX4 starting up by the usual beeps.
+
+:::note
+The file will be renamed to **autopilot.px4.updated** if the update was successful.
+If not, you should find a file **autopilot_update.log** with more details.
+:::
+
+### Update RC
+
+1. Place the **update.lzo** file on a USB stick (or SD card in a USB card reader). It needs to be FAT32 formatted.
+2. Plug the USB stick into the USB-A (not USB-C) port of the RC controller.
+3. Power on the RC controller by long pressing the power button.
+4. Wait for the blue LED to be on or blinking. This means the controller has finished booting.
+5. Hold the Home/RTL button (on the left) pressed while pressing the photo button (top right) quickly 4 times (within 2 seconds).
+6. Release the buttons when you hear a beep. It will beep while applying the update.
+7. Once the beeping stops with a positive tone (after about 90 seconds), the controller reboots automatically.
+
+:::note
+The **update.lzo** file is not remove on the USB stick after the update.
+:::
+
+## Developing PX4 Firmware
+
+### Build the PX4 Firmware
+
+To build PX4 for the Mantis, use the make command:
+```
+make atl_mantis-edu
+```
+
+### Upload/flash the PX4 Firmware
+
+There are two ways to update the firmware:
+
+#### Upload while connected to the Mantis' WiFi:
+
+The firmware will be uploaded to the SD-card over http using curl:
+```
+make atl_mantis-edu upload_wifi
+```
+
+:::note
+Make sure to have an SD card inserted for this.
+:::
+
+#### Upload using the SD card
+
```

Comment:
Add?

:::warning
The update method using the SD card relies on the camera being able to reboot the FMU via mavlink command. Should you flash a px4 firmware, where mavlink is not properly configured or booting, you might be unable to flash another firmware update using the SD card method.

TODO: What to do in case this happens?
:::

Reply:
Good Idea!

Reply:
Is this resolved?

Reply:
Yes, Julian's update actually prevents exactly this case from happening :)